### PR TITLE
its a pull request

### PR DESCRIPTION
**Pull request by aammd**
Just a few ideas about the analysis and code! very much a work in progress, no need to merge this with your stuff just yet!

### CODE REVIEW

File: figs/stanplots/analyses_wanang.R [hunk 1]
```diff
@@ -1,31 +1,43 @@
...
 # https://www.sciencedirect.com/science/article/pii/S0095447017302310
 
-# Upload data
-AllTestData <- read.table("all_test_data.txt")
-TreeTestDataset <- read.table("tree_test_data.txt")
+
+# load data and packages --------------------------------------------------
+
+# load data
+AllTestData <- read.table("figs/stanplots/all_test_data.txt")
```
Comment (aammd) [inline figs/stanplots/analyses_wanang.R]: it isn't useful to use setwd() when you have Rstudio projects! now the working directory is automatically set to where the .Rproj file lives (at the highest level of the directory) and we can specify a relative path from there

File: figs/stanplots/analyses_wanang.R [hunk 2]
```diff
@@ -1,31 +1,43 @@
...
 
 # Load package 
 library(brms)
+library(ggplot2)
+library(bayesplot)
+library(tidybayes)
+library(dplyr)
 
+
+# plots of the raw data ---------------------------------------------------
```
Comment (aammd) [inline figs/stanplots/analyses_wanang.R]: this is called a code section in rstudio! any comment that ends with multiple dashes. you can actually collapse them, and navigate among them! insert them with Cntrl-shift-r or Code>insert section

File: figs/stanplots/analyses_wanang.R [hunk 3]
```diff
@@ -1,31 +1,43 @@
...
+
+ggplot(AllTestData, aes(y = TREAT, x = BIO)) + 
+  geom_point() + 
+  facet_wrap(~GARDEN)
```
Comment (aammd) [inline figs/stanplots/analyses_wanang.R]: i couldn't think about data without ggplot2

File: figs/stanplots/analyses_wanang.R [hunk 4]
```diff
@@ -34,9 +46,121 @@ l1W_ranef <- brm(log(BIO) ~ TREAT + (1+TREAT|GARDEN),
...
+#contrasts(AllTestData$TREAT) <- contr.helmert(n = 6)
+
+bio_lnorm_bf <- bf(BIO ~ 1 + TREAT + (1 + TREAT|GARDEN), family = lognormal())
+brms::get_prior(bio_lnorm_bf, data = AllTestData)
+## Set some priors 
+priors <- c(set_prior("normal(4, 2)", class = "Intercept"),
+            set_prior("normal(0, 3)", class = "b"),
+            set_prior("normal(0, 4)", class = "sd"),
+            set_prior("normal(0, 3)", class = "sigma"),
+            set_prior("lkj(2)", class = "cor"))
```
Comment (aammd) [inline figs/stanplots/analyses_wanang.R]: These priors are a lot tighter than the ones that you set, but a) they can be played with b) they're still _very_ broad.

File: figs/stanplots/analyses_wanang.R [hunk 5]
```diff
@@ -34,9 +46,121 @@ l1W_ranef <- brm(log(BIO) ~ TREAT + (1+TREAT|GARDEN),
...
+brms::parnames(l1W_ranef)
+#get the info on divergent iterations
+np <- nuts_params(l1W_ranef)
+# customize appearance of divergences
+color_scheme_set("darkgray")
+div_style <- parcoord_style_np(div_color = "orange", div_size = 1, div_alpha = 1)
+mcmc_parcoord(as.array(l1W_ranef),
+              size = 0.25, alpha = 0.1,
+              regex_pars = "sd_.*",
+              np = np, np_style = div_style)+ coord_flip()
```
Comment (aammd) [inline figs/stanplots/analyses_wanang.R]: This is a technique for looking at those "divergent iterations" which is supposed to help you.. think about your model? its not straightforward

File: figs/stanplots/analyses_wanang.R [hunk 6]
```diff
@@ -34,9 +46,121 @@ l1W_ranef <- brm(log(BIO) ~ TREAT + (1+TREAT|GARDEN),
...
+
+# weevils as a number ------------------------------------------------------
+
+AllTestData_numweevil <- AllTestData %>% 
+  # could use relevel here
+  mutate(removal_trt = if_else(TREAT %in% c("CONTROL", "WEEVIL125", "WEEVIL25"), true = "0_removal", false = as.character(TREAT)),
+         removal_trt = as.factor(removal_trt),
+         weevil = readr::parse_number(TREAT)) %>% 
+  tidyr::replace_na(list(weevil = 0)) %>% 
+  mutate(weevil = weevil / 25)
```
Comment (aammd) [inline figs/stanplots/analyses_wanang.R]: Here is the conversion of weevil into numbers! and then scaling it to be a modest number. i don know if you use dplyr, but this is one (clumsy!) example of when it is useful!

File: figs/stanplots/analyses_wanang.R [hunk 7]
```diff
@@ -34,9 +46,121 @@ l1W_ranef <- brm(log(BIO) ~ TREAT + (1+TREAT|GARDEN),
...
+
+
+#perhaps not necessary, since using lognormal distribution seems to make a big difference.
+
+
+# weevils as a number ------------------------------------------------------
+
+AllTestData_numweevil <- AllTestData %>% 
+  # could use relevel here
+  mutate(removal_trt = if_else(TREAT %in% c("CONTROL", "WEEVIL125", "WEEVIL25"), true = "0_removal", false = as.character(TREAT)),
```
Comment (aammd) [inline figs/stanplots/analyses_wanang.R]: this could be forcats::relevel

File: figs/stanplots/analyses_wanang.R (hunk 5 above)
Comment (szefer-piotr) [reply figs/stanplots/analyses_wanang.R]: hmmm

**Comment by szefer-piotr** [comment]
This is great. Thank you so much!

**Pull request merged**